specifyweb/frontend/js_src/lib/components/LocalityUpdate/types.ts

```diff
@@ -0,0 +1,101 @@
+import type { State } from 'typesafe-reducer';
+
+import type { IR, RA } from '../../utils/types';
+import type { Tables } from '../DataModel/types';
+
+export type LocalityUpdateHeader = Exclude<
+  Lowercase<
+    | keyof Tables['GeoCoordDetail']['fields']
+    | keyof Tables['Locality']['fields']
+  >,
+  'locality'
+>;
+
+export type LocalityUpdateParseError = {
+  readonly message: string;
+  readonly field: string;
+  readonly payload: IR<unknown>;
+  readonly rowNumber: number;
+};
+
+export type LocalityUpdateTaskStatus =
```

Comment:
no need to duplicate it. can infer it

```suggestion
export type LocalityUpdateTaskStatus = LocalityUpdateState['state'];
```